src/main/java/reactor/rabbitmq/ChannelProxy.java

```diff
@@ -0,0 +1,580 @@
+        DeliverCallback deliverCallback, CancelCallback cancelCallback) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String basicConsume(String queue, boolean autoAck, String consumerTag, boolean noLocal, boolean exclusive, Map<String, Object> arguments,
+        DeliverCallback deliverCallback, ConsumerShutdownSignalCallback shutdownSignalCallback) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String basicConsume(String queue, boolean autoAck, String consumerTag, boolean noLocal, boolean exclusive, Map<String, Object> arguments,
+        DeliverCallback deliverCallback, CancelCallback cancelCallback, ConsumerShutdownSignalCallback shutdownSignalCallback) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void basicCancel(String consumerTag) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public AMQP.Basic.RecoverOk basicRecover() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public AMQP.Basic.RecoverOk basicRecover(boolean requeue) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public AMQP.Tx.SelectOk txSelect() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public AMQP.Tx.CommitOk txCommit() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public AMQP.Tx.RollbackOk txRollback() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public AMQP.Confirm.SelectOk confirmSelect() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long getNextPublishSeqNo() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean waitForConfirms() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean waitForConfirms(long timeout) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void waitForConfirmsOrDie() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void waitForConfirmsOrDie(long timeout) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void asyncRpc(Method method) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Command rpc(Method method) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long messageCount(String queue) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long consumerCount(String queue) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Command> asyncCompletableRpc(Method method) throws IOException {
+        if (!delegate.isOpen()) {
```

Comment:
Resource leak problem?
If 2 threads create new channel only one would be set to field, used and closed.

Reply:
Yes, you're right. We need to synchronize this.